doc/classes/SurfaceTool.xml

```diff
@@ -47,7 +48,8 @@
 			<argument index="0" name="index" type="int">
 			</argument>
 			<description>
-				Adds an index to index array if you are using indexed vertices. Does not need to be called before adding vertices.
+				Adds an index to index array if you are using indexed vertices.
+				[b]Note:[/b] If you call [method add_index], it must be called [i]before[/i] calling [method add_vertex]. Otherwise, it won't behave as desired.
```

Comment:
The index array can be created after finishing with all vertices or while vertices are being submitted.